rfd/0026/README.md

```diff
@@ -1274,38 +1289,36 @@ an appropriate error message when used on shared volumes zones.
 Machines acting as shared volumes' storage zones will have the value `nfsserver`
 for their `smartdc_role` property. To know of this new `smartdc_role` value is used by CloudAPI to prevent users from performing operations on these storage VMs, refer to
 
-#### New internal `required_nfs_volumes` property on VM objects
+#### New `volumes` property on VM objects
 
 A VM object that represents a Docker container mounting a shared volumes will
 store a reference to that volume in a new _indexable_ property named
-`required_nfs_volumes`.
+`volumes`.
 
 This will allow VOLAPI to perform an indexed search on VMAPI to determine what
 uses a given volume when handling operations that depends on a volume's usage,
 e.g a `DeleteVolume` request.
 
-This property is _internal_ and is not exposed to VMAPI and CloudAPI users. It
-is used by VMAPI's `ListVms` endpoint to implement support for its new
-[`mounting_volume` input
-parameter](#new-mounting_volume-parameter-for-the-listvms-endpoint).
+This property will also be exposed both at VMAPI and CloudAPI where it can be
+passed in as part of a VM payload to mount specified volumes on VM creation,
+and also can be listed via the ListVms and ListMachines endpoints.
 
-#### New `mounting_volume` parameter for the `ListVms` endpoint
+#### New `volumes` parameter for the `ListVms` endpoint
```

Comment:
This is a very good question. I looked to try to find some prior art here, but ListVms currently accepts only 2 lists `fields`, and `uuids` which work slightly differently. The `uuids` appears to be an *or* match, since any VMs in the list will be included in the output. The `fields` is more of a filter since it just specifies which fields are in the result.

Tags might be the closest because you can specify:

```
/vms?tag.role=sdc&tag.type=database
```

but the documentation is not clear whether this is an `and` or an `or`. It appears from some testing that this is however an `and`. To me it seems like `and` would be what I'd expect, if we did support multiple volumes. But I'd be happy to hear other opinions.